common-test/src/main/scala/org/genivi/sota/data/DeviceGenerators.scala

```diff
@@ -52,20 +54,13 @@ trait DeviceGenerators {
     genConflictFreeDeviceTs(arbitrary[Int].sample.get)
 
   def genConflictFreeDeviceTs(n: Int): Gen[Seq[DeviceT]] = {
-    val names: Seq[DeviceName] =
-      Gen.containerOfN[Seq, DeviceName](n, genDeviceName)
-      .suchThat { c => c.distinct.length == c.length }
-      .sample.get
-    val ids: Seq[DeviceId] =
-      Gen.containerOfN[Seq, DeviceId](n, genDeviceId)
-      .suchThat { c => c.distinct.length == c.length }
-      .sample.get
-
-    val namesG: Seq[Gen[DeviceName]] = names.map(Gen.const(_))
-    val idsG: Seq[Gen[DeviceId]] = ids.map(Gen.const(_))
-
-    namesG.zip(idsG).map { case (nameG, idG) =>
-      genDeviceTWith(nameG, idG).sample.get
+    for {
```

Comment:
Yes, the ones in the list should be conflict free, however, the suchThat() call can fail, so sometimes it would produce a None Option, and the .get() would subsequently fail. My for code tries to avoid this. I still feel I don't understand enough about what's going on to eliminate these issues entirely. I was gonna have a look at it later, since I felt I was spending too much time trying to get it working.